cpu/cc2538/radio/cc2538_rf_getset.c

```diff
@@ -62,60 +62,29 @@ static const uint8_t power_lut[NUM_POWER_LEVELS] = {
     255, /**<   7 dBm */
 };
 
-uint64_t cc2538_get_addr_long(void)
+void cc2538_get_addr_long(uint8_t *addr)
 {
-    uint64_t addr = RFCORE_FFSM_EXT_ADDR0;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR1;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR2;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR3;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR4;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR5;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR6;
-    addr <<= 8;
-    addr |= RFCORE_FFSM_EXT_ADDR7;
-    return addr;
+    addr[0] = RFCORE_FFSM_EXT_ADDR0;
+    addr[1] = RFCORE_FFSM_EXT_ADDR1;
+    addr[2] = RFCORE_FFSM_EXT_ADDR2;
+    addr[3] = RFCORE_FFSM_EXT_ADDR3;
+    addr[4] = RFCORE_FFSM_EXT_ADDR4;
+    addr[5] = RFCORE_FFSM_EXT_ADDR5;
+    addr[6] = RFCORE_FFSM_EXT_ADDR6;
+    addr[7] = RFCORE_FFSM_EXT_ADDR7;
 }
 
-uint16_t cc2538_get_addr_short(void)
+void cc2538_get_addr_short(uint8_t *addr)
 {
-    return (RFCORE_FFSM_SHORT_ADDR0 << 8) | RFCORE_FFSM_SHORT_ADDR1;
+    addr[0] = RFCORE_FFSM_SHORT_ADDR0;
+    addr[1] = RFCORE_FFSM_SHORT_ADDR1;
```

Comment:
```suggestion
    addr[1] = RFCORE_FFSM_SHORT_ADDR0;
    addr[0] = RFCORE_FFSM_SHORT_ADDR1;
```

to keep the same short.

Reply:
Yea then it mirrors `cc2538_set_addr_short()`

But I just noticed that the same issue must be present with `cc2538_get_addr_long()` / `cc2538_set_addr_long()` as there I do

    addr[0] = RFCORE_FFSM_EXT_ADDR0;

and

    RFCORE_FFSM_EXT_ADDR0 = addr[7];